starter-code/src/viking.js

```diff
@@ -1,11 +1,94 @@
 // Soldier
-function Soldier () {}
+function Soldier (healthParam, strengthParam) {
+  this.health = healthParam;
+  this.strength = strengthParam;
+}
 
+Soldier.prototype.attack = function(){
+  return this.strength;
+}
+
+Soldier.prototype.receiveDamage = function(damage){
+  this.health -= damage;
+}
 // Viking
-function Viking () {}
+function Viking(nameParam, healthParam, strengthParam) {
+  Soldier.call(this, healthParam, strengthParam);
+  this.name = nameParam;
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+
+Viking.prototype.receiveDamage = function(damage){
+  this.health -= damage;
+  if(this.health > 0)
+    return this.name + " has received " + damage + " points of damage";
+  else
+    return this.name + " has died in act of combat";
+}
 
+Viking.prototype.battleCry = function(){
+  return "Odin Owns You All!";
+}
 // Saxon
-function Saxon () {}
+function Saxon (healthParam, strengthParam) {
+  Soldier.call(this, healthParam, strengthParam);
+}
 
+Saxon.prototype = Object.create(Soldier.prototype);
+
+Saxon.prototype.receiveDamage = function(damage){
+  this.health -= damage;
+  if(this.health > 0)
+    return "A Saxon has received " + damage + " points of damage";
+  else if(this.health <= 0)
+    return "A Saxon has died in combat";
+}
 // War
-function War () {}
+function War () {
+  this.vikingArmy = [];
+  this.saxonArmy = [];
+}
+
+War.prototype.addViking = function(oneViking){
+  this.vikingArmy.push(oneViking);
+};
+
+War.prototype.addSaxon = function(oneSaxon){
+  this.saxonArmy.push(oneSaxon);
+};
+
+War.prototype.vikingAttack = function(){
+  var oneViking = this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)];
```

Comment:
Once you get more experience and confidence with your coding skills you will find that you can link together many operations and achieve what some would have done in multiple lines by just writting one line of code. 

Great job applying some advanced understanding of programming with these two lines. (line 62 and 63).